tareas/RuizGarcia/Tarea2Diego.jl

```diff
@@ -0,0 +1,642 @@
+# # Tarea 2: Exponentes de Lyapunov y universalidad
+
+#-
+# > Fecha de envío: 2 de julio
+# >
+# > Fecha de aceptación: 30 de julio
+
+#-
+# ## Ejercicio 1:
+#
+# Para la familia de mapeos cuadrática $Q_c(x) = x^2 + c$, generen el diagrama que
+# muestra el exponente de Lyapunov en términos de $c$, para $c\in[-2,2]$.
+# Utilizen un paso suficientemente fino (en $c$) para que el diagrama muestren
+# la riqueza del comportamiento.
+#
+# - ¿Qué particularidad ocurre (en términos del exponente de Lyapunov) para los valores de $c$ donde hay bifurcaciones?
+#
+# - ¿Qué particularidad ocurre (en términos del exponente de Lyapunov) cuando tenemos *ciclos superestables*, es decir, cuando $x=0$ es parte de un ciclo periódico?
+#
+
+#-
+# ### Respuesta:
+
+
+#-
+using TaylorSeries
+using Plots
+
+function derf(f,a) ## Defino una función que calcula la derivad de una función empleando TaylorSeries.
+    t = Taylor1(eltype(a), 1)
+    fT = f(a+t)
+    return fT[1]
+end
+
+## Defino el mapeo cuadrático.
+Q²(x,c)=x^2+c
+
+
+
+
+
+#-
+function lyapunov(f,n,x0) ## Primero defino una función que calcula el exponente de Lyapunov.
+    s=0
+    xs=ones(Float64,n+1)
+    xs[1]=x0
+    for i in 0:n-1
```

Comment:
*Sugerencia de programación*

```suggestion
    @inbounds for i in 0:n-1
```
esto hace más rápido el código cuando dentro del for entras a los arrays de la forma xs[i+1], es decir, si quieres usar tu código tal y como está esto puede ser más rápido ya que este @inbounds evita tener que verificar que el array en efecto tenga un índice i+1.

Reply:
Hola. Ya corregí esta parte del código. No obstante, me quedó una duda:
¿El compilador no optimiza esto de andar revisando el tamaño del array?

Reply:
Por defecto Julia tiene un nivel de seguridad en este tipo de cosas que verifica que lo que haces al menos existe, al poner @inbounds el compilador olvida esa parte y lo hace más eficaz.